controllers/checlusterbackup/internal_backup_server.go

```diff
@@ -264,21 +264,27 @@ func getInternalBackupServerConfigurationSpec(bctx *BackupContext) *chev1.CheBac
 			Labels:    labels,
 		},
 		Spec: chev1.CheBackupServerConfigurationSpec{
-			Rest: getExpectedInternalRestServerConfiguration(),
+			Rest: getExpectedInternalRestServerConfiguration(bctx),
 		},
 	}
 }
 
-func getExpectedInternalRestServerConfiguration() *chev1.RestServerConfig {
+func getExpectedInternalRestServerConfiguration(bctx *BackupContext) *chev1.RestServerConfig {
 	return &chev1.RestServerConfig{
 		Protocol:                    "http",
-		Hostname:                    backupServerServiceName,
+		Hostname:                    getInternalRestBackupServerUrl(bctx),
 		Port:                        backupServerPort,
 		RepositoryPath:              "che",
 		RepositoryPasswordSecretRef: BackupServerRepoPasswordSecretName,
 	}
 }
 
+// getInternalRestBackupServerUrl returns full service URL to be able to access it even from a different namespace.
+// This is needed for all namespaces operator mode.
+func getInternalRestBackupServerUrl(bctx *BackupContext) string {
+	return fmt.Sprintf("http://%s.%s.svc.cluster.local:%d", backupServerServiceName, bctx.namespace, backupServerPort)
```

Comment:
```suggestion
	return fmt.Sprintf("http://%s.%s.svc:%d", backupServerServiceName, bctx.namespace, backupServerPort)
```

Reply:
If internal communication is disabled we have to use public urls.

Reply:
Ok, thank you. Will rework it.